src/lib.rs

```diff
@@ -442,31 +452,29 @@ impl AstarteDeviceSdk {
         debug_assert!(interface.is_property());
         debug_assert!(interface.mapping(path).is_some());
 
-        //if database is loaded
-        if let Some(database) = &self.database {
-            let interface_name = interface.interface_name();
-            let version_major = interface.version_major();
+        let interface_name = interface.interface_name();
+        let version_major = interface.version_major();
 
-            database
-                .store_prop(interface_name, path.as_str(), data, version_major)
-                .await?;
+        self.store
+            .store_prop(interface_name, path.as_str(), data, version_major)
+            .await?;
 
-            // database selftest / sanity check for debug builds
-            if cfg!(debug_assertions) {
-                let stored_prop = database
-                    .load_prop(interface_name, path.as_str(), version_major)
-                    .await
-                    .expect("load_prop failed")
-                    .expect("property wasn't correctly saved in the database");
-
-                assert_eq!(*data, stored_prop);
-                let prop = self
-                    .property(interface_name, path)
-                    .await?
-                    .expect("property wasn't correctly saved in the database");
-                assert_eq!(*data, prop);
-                trace!("database test ok");
-            }
+        // storage self-test / sanity check for debug builds
+        if cfg!(debug_assertions) {
+            let stored_prop = self
+                .store
+                .load_prop(interface_name, path.as_str(), version_major)
+                .await
+                .expect("load_prop failed")
+                .expect("property wasn't correctly saved in the storage");
+
+            assert_eq!(*data, stored_prop);
+            let prop = self
+                .property(interface_name, path)
+                .await?
+                .expect("property wasn't correctly saved in the storage");
+            assert_eq!(*data, prop);
+            trace!("storage test ok");
```

Comment:
I think so too, it is a simple test to see if a property has been stored correctly, so it can be accomplished by a unit test.

Reply:
This is a check at runtime that would yield little value in the unit test. Since the device is already tested, but could catch some bugs during development. Maybe if we separate it in a different function?